File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort()
        answer = 0
        for i in range(k):
            _max = happiness.pop()
            if _max >= i:
                answer += _max - i
        return answer

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.rs
use std::collections::BinaryHeap;
impl Solution {
    pub fn maximum_happiness_sum(happiness: Vec<i32>, k: i32) -> i64 {
        let mut h = BinaryHeap::from(happiness);
        let mut answer = 0;
        for i in 0..k {
            if let Some(max) = h.pop() {
                if max >= i {
                    answer += (max - i) as i64;
                }
            }
        }
        answer
    }
}